Lombiq.Tests.UI/SecurityScanning/YamlDocumentExtensions.cs

```diff
@@ -0,0 +1,405 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using YamlDotNet.RepresentationModel;
+
+namespace Lombiq.Tests.UI.SecurityScanning;
+
+public static class YamlDocumentExtensions
+{
+    /// <summary>
+    /// Sets the start URL under the app where to start the scan from in the current context of the ZAP Automation
+    /// Framework plan.
+    /// </summary>
+    /// <param name="startUri">The absolute <see cref="Uri"/> to start the scan from.</param>
+    /// <exception cref="ArgumentException">
+    /// Thrown when the ZAP Automation Framework plan contains more than a single URL in the "urls" section.
+    /// </exception>
+    public static YamlDocument SetStartUrl(this YamlDocument yamlDocument, Uri startUri)
+    {
+        // Setting includePaths in the context is not necessary because by default everything under "urls" will be
+        // scanned.
+
+        var urls = yamlDocument.GetUrls();
+        var urlsCount = urls.Count();
+
+        if (urlsCount > 1)
```

Comment:
Is this due to a ZAP limitation? If I call AddUrl first and then call SetStartUrl, that will cause an error. Some URLs need to be added manually, if they aren't accessible via links, which might be case with registration, password reset, etc.